pkg/controller/garbagecollector/references/references.go

```
@@ -0,0 +1,153 @@
+		o.Annotations = utils.MergeStringMaps(o.Annotations, referenceAnnotations)
+		o.Spec.Template.Annotations = utils.MergeStringMaps(o.Spec.Template.Annotations, referenceAnnotations)
+
+	case *appsv1.StatefulSet:
+		referenceAnnotations := computeAnnotations(o.Spec.Template.Spec, additional...)
+		o.Annotations = utils.MergeStringMaps(o.Annotations, referenceAnnotations)
+		o.Spec.Template.Annotations = utils.MergeStringMaps(o.Spec.Template.Annotations, referenceAnnotations)
+
+	case *appsv1.DaemonSet:
+		referenceAnnotations := computeAnnotations(o.Spec.Template.Spec, additional...)
+		o.Annotations = utils.MergeStringMaps(o.Annotations, referenceAnnotations)
+		o.Spec.Template.Annotations = utils.MergeStringMaps(o.Spec.Template.Annotations, referenceAnnotations)
+
+	case *batchv1.Job:
+		referenceAnnotations := computeAnnotations(o.Spec.Template.Spec, additional...)
+		o.Annotations = utils.MergeStringMaps(o.Annotations, referenceAnnotations)
+		o.Spec.Template.Annotations = utils.MergeStringMaps(o.Spec.Template.Annotations, referenceAnnotations)
+
+	case *batchv1.CronJob:
+		referenceAnnotations := computeAnnotations(o.Spec.JobTemplate.Spec.Template.Spec, additional...)
+		o.Annotations = utils.MergeStringMaps(o.Annotations, referenceAnnotations)
+		o.Spec.JobTemplate.Annotations = utils.MergeStringMaps(o.Spec.JobTemplate.Annotations, referenceAnnotations)
+		o.Spec.JobTemplate.Spec.Template.Annotations = utils.MergeStringMaps(o.Spec.JobTemplate.Spec.Template.Annotations, referenceAnnotations)
+
+	case *batchv1beta1.CronJob:
+		referenceAnnotations := computeAnnotations(o.Spec.JobTemplate.Spec.Template.Spec, additional...)
+		o.Annotations = utils.MergeStringMaps(o.Annotations, referenceAnnotations)
+		o.Spec.JobTemplate.Annotations = utils.MergeStringMaps(o.Spec.JobTemplate.Annotations, referenceAnnotations)
+		o.Spec.JobTemplate.Spec.Template.Annotations = utils.MergeStringMaps(o.Spec.JobTemplate.Spec.Template.Annotations, referenceAnnotations)
+	}
```

return an error here if the given object GVK isn't handled. It's better to fail explicitly than do nothing silently.

---

Not sure, I would like to avoid error handling for the gazillion places this function will be used. The doc string already states what the function does, so passing an object which does not have any pod template spec does not make sense anyways.